client/src/components/DropdownSelect.tsx

```diff
@@ -0,0 +1,102 @@
+import { useState } from 'react'
+import { Listbox } from '@headlessui/react'
+import { LuChevronsUpDown, LuRefreshCcw, LuFile } from 'react-icons/lu';
+
+interface File {
+  name: string;
+  icon?: string;
+}
+
+const DropdownSelect = ({
+  label,
+  options,
+  refresh,
+}: {
+  label: string;
+  options: File[];
+  refresh: () => void;
+}) => {
+  const [, setSelected] = useState(options[0]);
+  const [searchItem, setOptions] = useState('');
+  const placeholder : File = {
+    name: label,
+  };
+
+  // filter out options that match the search string
+  const filteredOptions = options.filter((option) => {
+    return option.name.toLowerCase().includes(searchItem.toLowerCase())
+  });
+
+  return (
+    <div>
+      <button
+        className='flex items-center text-sm text-neutral-450'
```

Comment:
If you check this with axe or another accessibility checker, it'll show that this has insufficient color contrast.